svc/shows/transport.go

```diff
@@ -119,3 +190,98 @@ func codeAndMessageFrom(err error) (int, interface{}) {
 	}
 	return httpencoder.CodeAndMessageFrom(err)
 }
+
+func decodeAddShowRequest(_ context.Context, r *http.Request) (interface{}, error) {
+	b, err := strconv.ParseBool(r.URL.Query().Get(hasNewEpisode))
+	if err != nil {
+		return nil, fmt.Errorf("can not parse boolean value from string")
+	}
+
+	return AddShowRequest{
+		Title:         r.URL.Query().Get(title),
+		Cover:         r.URL.Query().Get(cover),
+		HasNewEpisode: b,
+		Category:      r.URL.Query().Get(category),
+	}, nil
+}
+
+func decodeUpdateShowRequest(_ context.Context, r *http.Request) (interface{}, error) {
+	b, err := strconv.ParseBool(r.URL.Query().Get(hasNewEpisode))
+	if err != nil {
+		return nil, fmt.Errorf("can not parse boolean value from string")
+	}
+
+	return UpdateShowRequest{
+		ID:            chi.URLParam(r, "show_id"),
+		Title:         r.URL.Query().Get(title),
+		Cover:         r.URL.Query().Get(cover),
+		HasNewEpisode: b,
+		Category:      r.URL.Query().Get(category),
+	}, nil
+}
+
+func decodeDeleteShowByIDRequest(_ context.Context, r *http.Request) (interface{}, error) {
+	id := chi.URLParam(r, "show_id")
+	if id == "" {
+		return nil, fmt.Errorf("%w: missed show_id", ErrInvalidParameter)
+	}
+	return id, nil
+}
+
+func decodeAddEpisodeRequest(_ context.Context, r *http.Request) (interface{}, error) {
+	return AddEpisodeRequest{
+		ShowID:        r.URL.Query().Get(showId),
+		EpisodeNumber: castStrToInt32(r.URL.Query().Get(episodeNumber)),
+		Cover:         r.URL.Query().Get(cover),
+		Title:         r.URL.Query().Get(title),
+		Description:   r.URL.Query().Get(description),
+		ReleaseDate:   r.URL.Query().Get(releaseDate),
+	}, nil
+}
+
+func decodeUpdateEpisodeRequest(_ context.Context, r *http.Request) (interface{}, error) {
+	return UpdateEpisodeRequest{
+		ID:            chi.URLParam(r, "id"),
+		ShowID:        r.URL.Query().Get(showId),
+		EpisodeNumber: castStrToInt32(r.URL.Query().Get(episodeNumber)),
+		Cover:         r.URL.Query().Get(cover),
+		Title:         r.URL.Query().Get(title),
+		Description:   r.URL.Query().Get(description),
+		ReleaseDate:   r.URL.Query().Get(releaseDate),
+	}, nil
+}
+
+func decodeDeleteEpisodeByIDRequest(_ context.Context, r *http.Request) (interface{}, error) {
+	id := chi.URLParam(r, "id")
+	if id == "" {
+		return nil, fmt.Errorf("%w: missed episode id", ErrInvalidParameter)
+	}
+	return id, nil
+}
+
+func decodeDeleteEpisodeByShowIDRequest(_ context.Context, r *http.Request) (interface{}, error) {
+	id := chi.URLParam(r, "show_id")
+	if id == "" {
+		return nil, fmt.Errorf("%w: missed show_id", ErrInvalidParameter)
+	}
+	return id, nil
+}
+
+func decodeGetEpisodeByIDRequest(_ context.Context, r *http.Request) (interface{}, error) {
+	id := chi.URLParam(r, "id")
+	if id == "" {
+		return nil, fmt.Errorf("%w: missed episode id", ErrInvalidParameter)
+	}
+
```

Comment:
I'd make same whitespace everywhere or would remove it here as well for consistency

Reply:
Do you mean spaces after if-else? @Qweder93